deploy/ocpbugs-48519/10-CronJob-4-17.yaml

```diff
@@ -0,0 +1,76 @@
+  namespace: openshift-pki-operator-annotator
+  annotations:
+    kubernetes.io/description: "Patches 4.17 ManifestWork to fixed pki-operator version https://issues.redhat.com/browse/OSD-27987"
+spec:
+  schedule: "*/5 * * * *" # Every five minutes
+  concurrencyPolicy: Replace
+  jobTemplate:
+    spec:
+      template:
+        metadata:
+          name: pki-operator-annotator
+          namespace: openshift-pki-operator-annotator
+          annotations:
+            kubernetes.io/description: "Patches 4.17 ManifestWork to fixed pki-operator version https://issues.redhat.com/browse/OSD-27987"
+        spec:
+          affinity:
+            nodeAffinity:
+              preferredDuringSchedulingIgnoredDuringExecution:
+                - preference:
+                    matchExpressions:
+                      - key: node-role.kubernetes.io/infra
+                        operator: Exists
+                  weight: 1
+          tolerations:
+            - effect: NoSchedule
+              key: node-role.kubernetes.io/infra
+              operator: Exists
+          containers:
+            - name: pki-operator-annotator
+              image: image-registry.openshift-image-registry.svc:5000/openshift/tools:latest
+              imagePullPolicy: Always
+              securityContext:
+                allowPrivilegeEscalation: false
+                capabilities:
+                  drop:
+                    - ALL
+                runAsNonRoot: true
+              env:
+                - name: IMAGE
+                  # 4.17.15
+                  value: "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:1ac8880940e256abb01cf8d8276beeb53db37572ab7369a752de5af244bc24e4"
+                - name: MAJOR_MINOR_VER
+                  value: "4.17"
+              command:
+                - /bin/bash
+              args:
+                - -c
+                - |
+                  # Get all manifestwork objects and extract their names
+                  managedclusters=$(oc get managedclusters -l openshiftVersion-major-minor=${MAJOR_MINOR_VER} -o jsonpath='{range .items[*]}{.metadata.name}{"\n"}{end}')
+                  # Loop through each manifestwork object
+                  for clusterID in ${managedclusters[@]};do 
+                    # Extract namespace and name
+                    namespace=$(oc get managedclusters $clusterID -o json | jq -r '.metadata.labels["api.openshift.com/management-cluster"]')
+                    kinds=$(oc get manifestwork -n $namespace $clusterID -o json | jq -r '.spec.workload.manifests[].kind')
+                    num=0 
+                    for kind in $kinds;do 
+                      if [[ $kind == "HostedCluster" ]]; then
+                          echo $clusterID
+                          #~1 escapes / in bash
+                          json_payload='[{"op":"replace","path":"/spec/workload/manifests/'"$num"'/metadata/annotations/hypershift.openshift.io~1control-plane-operator-image","value":"'"$IMAGE"'"}]'
```

Comment:
I don't do enough [JSON Patch][1] to have internalized this, so I checked the `~1` escaping, and [it's JSON Pointer for `/`][2], to set up the expected `hypershift.openshift.io/control-plane-operator-image` annotation 👍 

[1]: https://www.rfc-editor.org/rfc/rfc6902
[2]: https://www.rfc-editor.org/rfc/rfc6901#section-3

Reply:
Yeah this is just copied from another CronJob we have doing this same behavior, I did not look into it too much besides confirming it works as expected.